packages/vscode-extension/src/debug/teamsfxTaskProvider.ts

```diff
@@ -78,8 +78,69 @@ export class TeamsfxTaskProvider implements vscode.TaskProvider {
     task: vscode.Task,
     token?: vscode.CancellationToken | undefined
   ): Promise<vscode.Task | undefined> {
-    // Return undefined since all tasks are provided and fully resolved
-    return undefined;
+    // Resolve "npm run dev" and "npm run watch" tasks
+    if (vscode.workspace.workspaceFolders) {
+      const workspaceFolder: vscode.WorkspaceFolder = vscode.workspace.workspaceFolders[0];
+      const workspacePath: string = workspaceFolder.uri.fsPath;
+      if (!(await commonUtils.isFxProject(workspacePath))) {
+        vscodeLogger.error(
+          `No ${TeamsfxTaskProvider.type} project. Cannot resolve ${TeamsfxTaskProvider.type} task.`
+        );
+        return undefined;
+      }
+
+      const command: string | undefined = task.definition.command;
+      const component: string | undefined = task.definition.component;
+      if (
+        command &&
+        task.scope !== undefined &&
+        task.scope !== vscode.TaskScope.Global &&
+        task.scope !== vscode.TaskScope.Workspace
+      ) {
+        let env: { [key: string]: string } | undefined;
+        let cwd: string | undefined;
+        let problemMatcher: string;
+        const isWatchTask = /^npm run watch/i.test(command + "");
+        if (component?.toLowerCase() === "frontend") {
+          env = isWatchTask ? undefined : await commonUtils.getFrontendLocalEnv();
```

Comment:
So according to our definition, `npm run watch` is just a task to watch files (like `tsc -w`) but not a task to start service directly with files watched?

Reply:
Right, and we already have `watch` in ts function package.json (https://github.com/OfficeDev/TeamsFx/blob/dev/templates/function-base/ts/default/package.json#L6)

Actually we only need `watch` for backend `func`.